crates/arroyo-formats/src/de.rs

```diff
@@ -82,6 +83,9 @@ pub struct ArrowDeserializer {
     schema_registry: Arc<Mutex<HashMap<u32, apache_avro::schema::Schema>>>,
     proto_pool: DescriptorPool,
     schema_resolver: Arc<dyn SchemaResolver + Sync>,
+    enable_metadata: Option<bool>,
+    kafka_metadata_builder: Option<(Int64Builder, Int32Builder, StringBuilder)>,
```

Comment:
This approach is very specific to Kafka, but the goal for this project is to have a generic metadata system that any connector can opt into with its specific metadata fields. How can we make the deserialization support work for whatever metadata fields a connector has?